frontend/src/utils/TaskUtils.ts

```diff
@@ -220,51 +220,61 @@ export const dragDropBetweenLists = (
   };
 };
 
-export const calcTaskWeightedValueSum = (
-  task: Task,
-  allCustomers: Customer[],
-  roadmap: Roadmap,
+const ratingValue = (allCustomers: Customer[], roadmap: Roadmap) => (
+  rating: Taskrating,
 ) => {
-  const customerValuesSum = allCustomers.reduce(
-    (total, customer) => total + customerWeight(customer),
-    0,
+  const ratingCreator = allCustomers.find(
+    ({ id }) => id === rating.forCustomer,
   );
+  const creatorWeight = ratingCreator
+    ? customerWeight(ratingCreator, roadmap.plannerCustomerWeights)
+    : 0;
 
-  const ratingValue = (rating: Taskrating) => {
-    const ratingCreator = allCustomers.find(
-      ({ id }) => id === rating.forCustomer,
-    );
-
-    const ratingCreatorValue = ratingCreator
-      ? customerWeight(ratingCreator, roadmap.plannerCustomerWeights)
-      : 0;
-
-    let creatorPlannerWeight = roadmap.plannerCustomerWeights?.find(
-      ({ customerId }) => customerId === rating.forCustomer,
-    )?.weight;
-    if (creatorPlannerWeight === undefined) creatorPlannerWeight = 1;
-
-    let creatorValueWeight = 0;
-    if (ratingCreator === undefined) creatorValueWeight = 1;
-
-    if (ratingCreatorValue > 0 && customerValuesSum > 0) {
-      creatorValueWeight = ratingCreatorValue / customerValuesSum;
-    }
-    return rating.value * creatorValueWeight * creatorPlannerWeight;
-  };
+  return rating.value * creatorWeight;
+};
 
-  return task.ratings
+const taskWeightedValueSummary = (
+  result: RatingsSummary,
```

Comment:
The `result` parameter might be simpler to remove, and just create `new RatingsSummary()` in this function.

Or did you have some use case in mind, where the result would be needed as parameter?

Reply:
No, I can't think of a case when it would be needed. 
That's really only the side-effect of me not comprehending ratingsSummaryByDimensionInto and trying to mimic its usage. I now see the solution is actually simple... (and understand what exactly is going on in ratingsSummaryByDimensionInto)